### PR TITLE
Docker setup

### PR DESCRIPTION
**Pull request by DarshanAgarwal09**

### CODE REVIEW

File: DA/server.ts [hunk 1]
```diff
@@ -107,7 +108,21 @@ async function saveToDatabase(encTile: any) {
 
     socket.emit("saveToDatabaseResponse");
 }
-
+async function create_encrypted_tiles(){
```
Comment (lyronctk) [inline DA/server.ts:111]: natural language paragraph comment above each function please. one or two sentences describing what + why is plenty

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+FROM node as client
+RUN npm install -g pnpm@${PNPM_VERSION}
+
+WORKDIR /usr/src/app
+# COPY . .
+COPY --from=contracts /usr/src/app /usr/src/app
+RUN sh docker_scripts/client_setup.sh
+
+# Run the application.
+#CMD ["pnpm", "-C", "client", "devA"]
```
Comment (lyronctk) [inline Dockerfile:65]: nit: no left over comments in main

File: compose.yaml [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    volumes:
+      - common-contracts:/usr/src/app/contracts
+    depends_on:
+      enclave:
+        condition: service_healthy
+volumes:
+  db-data:
+    driver: local
+  common-contracts:
+    driver: local
```
Comment (lyronctk) [inline compose.yaml:75]: nit: newline

File: docker_scripts/client_setup.sh [hunk 4]
```diff
@@ -0,0 +1,7 @@
+cd game
+pnpm install
+cd ..
+
+cd client
+pnpm install
+cd ..
```
Comment (lyronctk) [inline docker_scripts/client_setup.sh:7]: nit: newline

File: docker_scripts/contracts_run.sh [hunk 5]
```diff
@@ -0,0 +1,7 @@
+rm -rf /usr/src/app/contracts/worlds.json
+
+. /root/.bashrc
+foundryup
+
+
+pnpm -C contracts dev
```
Comment (lyronctk) [inline docker_scripts/contracts_run.sh:7]: nit: newline

File: docker_scripts/contracts_setup.sh [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+cd artifacts
+wget https://storage.googleapis.com/zkevm/ptau/powersOfTau28_hez_final_17.ptau
+cd ..
+yarn dev
+rm -rf node_modules
+cd ..
+
+cd contracts
+pnpm build
+cd ..
```
Comment (lyronctk) [inline docker_scripts/contracts_setup.sh:17]: nit: newline

File: docker_scripts/da_setup.sh [hunk 7]
```diff
@@ -0,0 +1,4 @@
+
+cd DA
+pnpm install
+cd ..
```
Comment (lyronctk) [inline docker_scripts/da_setup.sh:4]: nit: newline

File: README.md [hunk 8]
```diff
@@ -1,18 +1,58 @@
...
+
+### Manual setup (WIP) 
+
+* Set up local network
 ``` 
 anvil
 ```
 
-### Compile circuit
+* Compile circuit
```
Comment (lyronctk) [inline README.md:48]: mind updating these instructions as well? :)

**Review by alex-lindenbaum — APPROVED** [review]
lgtm!

**Pull request merged**